static_frame/core/frame.py

```diff
@@ -5968,6 +6013,27 @@ def to_latex(self,
                 )
         return repr(self.display(config))
 
+    def to_msgpack(self) -> 'msgpack':
+        '''
+        Return a msgpack.
+        '''
+        import msgpack
+        import msgpack_numpy
+        def cast_msgpack(input):
+            try:
+                data = msgpack.packb(input, default=msgpack_numpy.encode) #try coercing standard and numpy datatypes
+            except TypeError:
+                #This inserts a magic character with any arbitrary datatype that msgpack_numpy doesn't support
+                #TODO: Is this attempting to handle too much? Is there a better way?
+                clsname = '>'+input[0].__class__.__module__+'.'+input[0].__class__.__name__                
+                data = msgpack.packb([clsname]+[a.__str__() for a in input], use_bin_type=True) #else cast to string
```

Comment:
Encoding the string form of each data element seems a bit inefficient: is there no other option? And, on import, will we get back the types we originally had? If I have an `Index` of integers, will we get back integers on import?

Reply:
Definitely inefficient. This is the last resort fallback for anything msgpack_numpy can't handle.

Reply:
The better option is to build custom mspack datatypes for every unsupported datatype that we need, but I think this is the best fallback option for making sure things still work when library users start injecting their own foreign data types.

Reply:
Perhaps handling datetime64 explicitly will remove the need for this.